client-java/controller/src/main/java/org/evomaster/client/java/controller/opensearch/utils/OpenSearchQueryHelper.java

```diff
@@ -0,0 +1,88 @@
+package org.evomaster.client.java.controller.opensearch.utils;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.IllegalAccessException;
+import java.lang.NoSuchMethodException;
+
+/**
+ * Helper class for working with OpenSearch queries.
+ */
+public class OpenSearchQueryHelper {
+    /**
+     * Extracts the kind of query (term, match, etc.) from a query object.
+     */
+    public static String extractQueryKind(Object query) {
+        try {
+            Object kind = query.getClass().getMethod("_kind").invoke(query);
+            return (String) kind.getClass().getMethod("name").invoke(kind);
+        } catch (IllegalAccessException | InvocationTargetException | NoSuchMethodException e) {
+            throw new RuntimeException(e);
+        }
+
+    }
+
+    /**
+     * Extracts the field name from a query object.
+     */
+    public static String extractFieldName(Object query, String structure) {
+        try {
+            Object term = query.getClass().getMethod(structure).invoke(query);
+            return (String) term.getClass().getMethod("field").invoke(term);
+        } catch (IllegalAccessException | InvocationTargetException | NoSuchMethodException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Extracts the field value from a query object with its t.
+     */
+    public static Object extractFieldValue(Object query, String structure) {
+        try {
+            Object term = query.getClass().getMethod(structure).invoke(query);
+            Object value = term.getClass().getMethod("value").invoke(term);
+            return extractTypedFieldValue(value);
+        } catch (IllegalAccessException | InvocationTargetException | NoSuchMethodException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Extracts the value from a field by its type (Double, Long, Boolean, String, Null).
+     */
+    public static Object extractTypedFieldValue(Object value) {
+        try {
+            Object kind = value.getClass().getMethod("_kind").invoke(value);
+            String kindName = (String) kind.getClass().getMethod("name").invoke(kind);
+
+            switch (kindName) {
```

Comment:
what about `Integer` and `Short`?

Reply:
I used OpenSearch [FieldValue](https://github.com/opensearch-project/opensearch-java/blob/main/java-client/src/main/java/org/opensearch/client/opensearch/_types/FieldValue.java#L74) as reference. Will review if more are required.